### PR TITLE
lntest: Use TCP ports not based in nodeID

### PR DESCRIPTION
**Pull request by matheusd**
This changes TCP port selection in integration tests from being
sequential, based on the node ID to being sequential but tested before
assigment.

This should reduce the number of flaky tests that fail due to the port
already being used by another process in the CI server.

Example of a test that failed due to this: https://travis-ci.org/lightningnetwork/lnd/jobs/612299358#L337

### CODE REVIEW

File: lntest/node.go [hunk 1]
```diff
@@ -43,31 +44,10 @@ const (
...
-	// ports for future rpc harness instances will be monotonically
-	// increasing numbers calculated as such:
-	// defaultProfilePort + (4 * harness.nodeNum).
-	defaultProfilePort = defaultNodePort + 3
-
-	// logPubKeyBytes is the number of bytes of the node's PubKey that
-	// will be appended to the log file name. The whole PubKey is too
-	// long and not really necessary to quickly identify what node
-	// produced which log file.
+	// logPubKeyBytes is the number of bytes of the node's PubKey that will
```
Comment (wpaulino) [inline lntest/node.go]: Nit: godoc of `defaultNodePort` should be updated.

File: lntest/node.go [hunk 2]
```diff
@@ -80,6 +60,10 @@ var (
 	numActiveNodes    = 0
 	numActiveNodesMtx sync.Mutex
 
+	// lastPort is the last port determined to be free for use by a new
+	// node. It should be used atomically.
```
[the remark this answers is not in the document]
Comment (matheusd) [reply lntest/node.go]: It's **atomically** (not automatically) to imply it must be only be accessed by `sync/atomic` functions (similar to, for example https://github.com/lightningnetwork/lnd/blob/master/discovery/syncer.go#L253-L255)

File: lntest/node.go [hunk 3]
```diff
@@ -91,19 +75,35 @@ var (
...
+// nextAvailablePort returns the first port that is available for listening by
+// a new node. It panics if no port is found and the maximum available TCP port
+// is reached.
+func nextAvailablePort() int {
+	port := atomic.AddUint32(&lastPort, 1)
+	for port < 65535 {
+		addr := fmt.Sprintf("127.0.0.1:%d", port)
+		l, err := net.Listen("tcp4", addr)
+		if err == nil {
+			l.Close()
```
Comment (halseth) [inline lntest/node.go]: should check error of this operation also

**Comment by halseth** [comment]
Good catch! Did you look into whether this fix is related to the fix in #3751?

**Comment by matheusd** [comment]
I don't think it is because nodes are not started concurrently in integration tests at the moment and I'm still getting this error occasionally even though I've already changed dcrlnd to use atomic node ids:

https://github.com/decred/dcrlnd/blob/master/lntest/node.go#L345-L347

So at least at this moment it seems to me that the issue is some other (non-lnd related) process running in the CI server.

**Comment by halseth** [comment]
Linter is failing

**Comment by wpaulino** [comment]
@matheusd looks good now, could you squash the fixup commits?

File: lntest/node.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (carlaKC) [reply lntest/node.go]: sorry, skim read this before morning coffee 😅

**Pull request merged**